src/map/mod.rs

```diff
@@ -0,0 +1,276 @@
+    View<State, Action, ViewCtx, MapMessage, Element = MapChildElement>
+{
+}
+
+impl<V, State, Action> MapChild<State, Action> for V where
+    V: View<State, Action, ViewCtx, MapMessage, Element = MapChildElement>
+{
+}
+
+pub struct Map<State, Action, Children> {
+    map_view: MapDomView<State, Action>,
+    children: Children,
+    zoom: Option<f64>,
+    center: Option<(f64, f64)>,
+    phantom: PhantomData<fn() -> (State, Action)>,
+}
+
+impl<State, Action, Children> Map<State, Action, Children> {
+    pub fn zoom(mut self, value: f64) -> Self {
+        self.zoom = Some(value);
+        self
+    }
+
+    pub fn on_zoom_end<F>(&self, callback: F) -> OnZoomEnd<State, F>
+    where
+        F: Fn(&mut State, f64) + 'static,
+    {
+        on_zoom_end(callback)
+    }
+
+    pub fn center(mut self, lat: f64, lng: f64) -> Self {
+        self.center = Some((lat, lng));
+        self
+    }
+}
+
+impl<State, Action, Children> ViewMarker for Map<State, Action, Children> {}
+
+pub struct MapViewState<DS, CS> {
+    map_dom_state: DS,
+    children_state: CS,
+    leaflet_map: leaflet::Map,
+}
+
+#[derive(Debug, Clone)]
+pub enum MapMessage {
+    InitMap,
+    MapHasMounted(leaflet::Map),
+    ZoomEnd(f64),
+}
+
+/// Distinctive id for better debugging
+const MAP_VIEW_ID: ViewId = ViewId::new(1236068);
+
+impl<State, Action, Children> View<State, Action, ViewCtx, DynMessage>
+    for Map<State, Action, Children>
+where
+    State: 'static,
+    Action: 'static,
+    Children: MapChildren<State>,
+{
+    type Element = <MapDomView<State, Action> as View<State, Action, ViewCtx, DynMessage>>::Element;
+
+    type ViewState = MapViewState<
+        <MapDomView<State, Action> as View<State, Action, ViewCtx, DynMessage>>::ViewState,
+        Children::SeqState,
+    >;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        ctx.with_id(MAP_VIEW_ID, |ctx| {
+            let (map_dom_element, map_dom_state) = self.map_view.build(ctx);
+
+            let map_options = leaflet::MapOptions::default();
+            let leaflet_map = leaflet::Map::new_with_element(&map_dom_element.node, &map_options);
+
+            let mut elements = AppendVec::default();
+            let children_state = self.children.seq_build(ctx, &mut elements);
+
+            let view_state = MapViewState {
+                leaflet_map,
+                map_dom_state,
+                children_state,
+            };
+
+            // Is the following an issue?
+            // Does it require on being in the DOM tree?
+            // Can the tile-layer be added before running `apply_zoom_and_center`?
+
+            // We have to postpone the map initiation
+            // because the DOM element has been created at this point in time
+            // but has not yet been mounted.
+            let thunk = ctx.message_thunk();
+            spawn_local(async move {
+                thunk.push_message(MapMessage::InitMap);
+            });
```

Comment:
Btw. that's an interesting case, maybe we can add some syntax sugar for this in general (like `onMount` at least), `teardown` is basically `beforeUnmount`. That said, I think we should better not rely on *reading* things from the DOM if possible (rather rely on event views such as the new `OnResize` with a `ResizeObserver`), e.g. because of layout-thrashing etc.